transactions/transactor_test.go

```diff
@@ -385,41 +401,152 @@ func (s *TxQueueTestSuite) TestLocalNonce() {
+				ensParsed, _ := abi.JSON(strings.NewReader(contract.ENSABI))
+				secondOwnerAddress := account.FromAddress(TestConfig.Account1.Address)
+				setOwnerMethodData, _ := ensParsed.Pack("setOwner", setOwnerMethodID, node, secondOwnerAddress)
+
+				// method ID: web3.utils.sha3('owner(bytes32)').substring(0, 8)
+				ownerMethodID := "0x02571b"
+				getOwnerMethodData, _ := ensParsed.Pack("owner", ownerMethodID, node)
+
+				return []SendTxArgs{
+					{
+						// Create contract
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+					{
+						// Change contract owner
+						Input: hexutil.Bytes(setOwnerMethodData),
+					},
+					{
+						// Check that the contract owner is correct
+						Input: hexutil.Bytes(getOwnerMethodData),
+					},
+				}
+			},
+			func(receipt *types.Receipt, finalResult sign.Result, testaddr common.Address) {
+				successStatus := uint(1)
+				s.Equal(successStatus, receipt.Status)
+				// secondOwnerAddress := account.FromAddress(TestConfig.Account1.Address)
+				// s.Equal(finalResult.Response.Hex(), secondOwnerAddress.Hex())
+			},
+		},
 	}
 
-	go func() {
-		for i := 1000; i > 0; i-- {
-			req := s.manager.pendingSignRequests.First()
-			if req == nil {
-				time.Sleep(time.Millisecond)
-			} else {
-				s.manager.pendingSignRequests.Approve(req.ID, "", s.defaultSignTxArgs(), simpleVerifyFunc(selectedAccount)) // nolint: errcheck
-				break
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+
+			key, _ := crypto.GenerateKey()
+			testaddr := crypto.PubkeyToAddress(key.PublicKey)
+			genesis := core.GenesisAlloc{
+				testaddr: {Balance: big.NewInt(100000000000)},
+			}
+			backend := backends.NewSimulatedBackend(genesis)
+			selectedAccount := &account.SelectedExtKey{
+				Address:    testaddr,
+				AccountKey: &keystore.Key{PrivateKey: key},
 			}
-		}
-	}()
 
-	hash, err := s.manager.SendTransaction(context.Background(), tx)
-	s.NoError(err)
-	backend.Commit()
-	receipt, err := backend.TransactionReceipt(context.TODO(), hash)
-	s.NoError(err)
-	s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
+			s.manager.sender = backend
+			s.manager.gasCalculator = backend
+			s.manager.pendingNonceProvider = backend
+
+			waiter := make(chan struct{})
+			defer func() {
+				<-waiter
+			}()
+			closer := make(chan struct{})
+			defer close(closer)
+
+			// finalResult corresponds to the last transaction
+			getFinalResult := make(chan sign.Result)
+			transactions := testCase.getTransactions(testaddr)
+			go func() {
+				numTransactions := 0
+				for {
+					select {
+					case <-closer:
+						close(waiter)
+						return
+					default:
+					}
+					req := s.manager.pendingSignRequests.First()
+					if req == nil {
+						time.Sleep(time.Millisecond)
+					} else {
+						numTransactions++
+						if numTransactions == len(transactions) {
+							go func() {
+								getFinalResult <- s.manager.pendingSignRequests.Wait(req.ID, time.Second*1)
```

Comment:
Here is where I'm trying to get the result from the sign request in order to read it below and pass it to the 'receiptValidator' function, which I should probably also rename. 

This simply is not returning at the moment. Do you know if the simulated backend even allows for that?